cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -627,16 +627,29 @@ static PyObject *set_signal_val_real(gpi_hdl_Object<gpi_sim_hdl> *self, PyObject
     Py_RETURN_NONE;
 }
 
+static PyObject *set_signal_val_int(gpi_hdl_Object<gpi_sim_hdl> *self, PyObject *args)
+{
+    int value;
+    gpi_set_action_t action;
+
+    if (!PyArg_ParseTuple(args, "ii:set_signal_val_int", &action, &value)) {
+        return NULL;
+    }
+
+    gpi_set_signal_value_int(self->hdl, static_cast<int32_t>(value), action);
+    Py_RETURN_NONE;
+}
+
 static PyObject *set_signal_val_long(gpi_hdl_Object<gpi_sim_hdl> *self, PyObject *args)
 {
-    long value;
+    long long value;
     gpi_set_action_t action;
 
-    if (!PyArg_ParseTuple(args, "il:set_signal_val_long", &action, &value)) {
+    if (!PyArg_ParseTuple(args, "iL:set_signal_val_long", &action, &value)) {
         return NULL;
     }
 
-    gpi_set_signal_value_long(self->hdl, value, action);
+    gpi_set_signal_value_int(self->hdl, static_cast<int32_t>(value), action);
```

Comment:
```suggestion
    if (value < std::numeric_limits<int32_t>::min()) {
        PyErr_SetString(PyExc_OverflowError, "value is less than minimum int32");
        return NULL;
    }
    else if (value > std::numeric_limits<int32_t>::max()) {
        PyErr_SetString(PyExc_OverflowError, "value is greater than maximum int32");
        return NULL;
    }
    gpi_set_signal_value_int(self->hdl, static_cast<int32_t>(value), action);
```

Reply:
Nice! I was just wondering how to do this exactly from C++.